plugins/org.komodo.shell/src/org/komodo/shell/commands/ResetGlobalPropertyCommand.java

```diff
@@ -0,0 +1,107 @@
+package org.komodo.shell.commands;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.komodo.shell.BuiltInShellCommand;
+import org.komodo.shell.CommandResultImpl;
+import org.komodo.shell.CompletionConstants;
+import org.komodo.shell.ShellI18n;
+import org.komodo.shell.api.CommandResult;
+import org.komodo.shell.api.WorkspaceStatus;
+import org.komodo.utils.StringUtils;
+import org.komodo.utils.i18n.I18n;
+
+public class ResetGlobalPropertyCommand extends BuiltInShellCommand {
+	/**
+	 * The command name.
+	 */
+	public static final String NAME = "reset-global"; //$NON-NLS-1$
+
+	private final static String ARG_ALL = "--all";
+
+	public ResetGlobalPropertyCommand(final WorkspaceStatus workspaceStatus) {
+		super(workspaceStatus, NAME);
+	}
+
+	@Override
+	public boolean isValidForCurrentContext() {
+		return true;
+	}
+
+	@Override
+	protected CommandResult doExecute() {
+		try {
+		
+			String firstArgument = requiredArgument(0, I18n.bind(ShellI18n.invalidArgMsgResetGlobalPropertyName,ARG_ALL));
+
+			WorkspaceStatus status = getWorkspaceStatus();
+			if (firstArgument.equals(ARG_ALL)) {
+				// reset all global properties
+				for (final Entry<String, String> entry : WorkspaceStatus.GLOBAL_PROPS.entrySet()) {
```

Comment:
Instead of resetting the properties the way you have done (which works), you can just call `WorkspaceStatus.setProperties(null)`.